crates/cxx-qt-gen/src/parser/cxxqtdata.rs

```diff
@@ -291,6 +292,15 @@ impl ParsedCxxQtData {
                 return Ok(None);
             // Find if we are an impl block for a qobject
             } else if let Some(qobject) = self.qobjects.get_mut(&path_to_single_ident(path)?) {
+                // Find if we are implemeting a trait for the QObject
+                if let Some((not, path, _)) = &imp.trait_ {
+                    // Find if we are enabling Threading
+                    if not.is_none() && path_compare_str(path, &["cxx_qt", "Threading"]) {
```

Comment:
I think this should be restructured slightly.
Check the path_compare_str() first.
Then add assertions that not is_none() and return an error otherwise (i.e. "Negative impls for cxx_qt::Threading not allowed").
Also, please assert that the actual implementation is empty.

Reply:
done